diff --git a/node_modules/tiny-secp256k1/lib/validate.js b/node_modules/tiny-secp256k1/lib/validate.js
index 9260a4f..d752d70 100644
--- a/node_modules/tiny-secp256k1/lib/validate.js
+++ b/node_modules/tiny-secp256k1/lib/validate.js
@@ -1,5 +1,5 @@
 const { instance } = require('./wasm');
-const { UINT32_MAX_BN } = require('./constants');
+const { UINT32_MAX_BN, FORCE_PULL_PURE_JS } = require('./constants');
 const BN = require('tiny-bn');
 
 function validatePoint(p) {
@@ -21,7 +21,7 @@
   return bn.isZero() || bn.gte(UINT32_MAX_BN) ? false : true;
 }
 
-module.exports = function (secp256k1) {
+const wrapper = function (secp256k1) {
   const {
     isPoint,
     isPointCompressed,
@@ -79,3 +79,9 @@
     isXOnlyPoint: isXOnlyPoint,
   };
 };
+
+if (FORCE_PULL_PURE_JS) {
+  module.exports = wrapper(require('./pure'));
+} else {
+  module.exports = wrapper(instance());
+}
diff --git a/node_modules/tiny-secp256k1/lib/constants.js b/node_modules/tiny-secp256k1/lib/constants.js
index 28e8331..d0c5a24 100644
--- a/node_modules/tiny-secp256k1/lib/constants.js
+++ b/node_modules/tiny-secp256k1/lib/constants.js
@@ -1,6 +1,9 @@
 const BN = require('tiny-bn');
 
+const FORCE_PULL_PURE_JS =
+  process.env.TINY_SECP256K1_FORCE_PUREJS === 'true';
 module.exports = {
   UINT32_MAX_BN: new BN(2).pow(new BN(32)),
   // compact signatures have 32 bytes of R, 32 bytes of S, and 1 byte of V
   EC_COMPACT_SIGNATURE_SIZE: 65,
+  FORCE_PULL_PURE_JS,
 };
diff --git a/node_modules/tiny-secp256k1/index.js b/node_modules/tiny-secp256k1/index.js
index c6b939f..1d67d26 100644
--- a/node_modules/tiny-secp256k1/index.js
+++ b/node_modules/tiny-secp256k1/index.js
@@ -1,5 +1,11 @@
-const { instance } = require('./lib/wasm');
-const validate = require('./lib/validate');
+const { FORCE_PULL_PURE_JS } = require('./lib/constants');
 
-const secp256k1 = instance();
-module.exports = Object.assign(secp256k1, validate(secp256k1));
+if (FORCE_PULL_PURE_JS) {
+  const secp256k1 = require('./lib/pure');
+  const validate = require('./lib/validate');
+  module.exports = Object.assign(secp256k1, validate(secp256k1));
+} else {
+  const { instance } = require('./lib/wasm');
+  const validate = require('./lib/validate');
+  const secp256k1 = instance();
+  module.exports = Object.assign(secp256k1, validate(secp256k1));
+}
